code/src/components/TodoForm.js

```diff
@@ -0,0 +1,40 @@
+import React, { useState }  from 'react';
+import { useDispatch } from 'react-redux';
+import uniqid from 'uniqid';
+
+import todos from '../reducers/todos';
+
+const TodoForm = () => {
+    const [value, setValue] = useState('');
+
+    const dispatch = useDispatch();
+
+    const onFormSubmit = (e) => {
+        e.preventDefault();
+
+        const newTodo = {
+            id: uniqid(),
+            description: value,
+            isComplete: false
+        };
+
+        dispatch(todos.actions.addTodo(newTodo));
+        setValue('');
+    }
+
+    return(
+        <>
+        <form className="form" onSubmit={onFormSubmit}>
+            <input className="input"
+               placeholder="Write your todo here..."
+               type="text"
+               value={value}
+               onChange={e => setValue(e.target.value)}
```

Comment:
FIGHT WITH LAZYNESS!! 🔫 add REQUIRED to make your to-do free from empty tasks!! 🎖️ 😆